types/index.d.ts

```diff
@@ -60,13 +60,34 @@ export interface ClipboardStatic {
     dangerouslyPasteHTML(index: number, html: string, source?: Sources): void;
 }
 
+/**
```

Comment:
> We can write the explanation about the registries inner workings here, as a JSDoc attached to the property (so it's accessible using IDE documentation tools).

Is this what you meant? Would IDE be able to pick it up this way? I'll add a link to this JSDoc on the front end, once we merged the changes

Reply:
You don't have to redefine every property type in JSDoc, those are already described in the type itself.
I think you should be able to just place the comment above the `registry` property.
```
/**
 * Description goes here
 * /
 registry?: Parchment.Registry;
```

In order to check whether or not the IDE picks that up, install the package locally and hover over the registry field in your editor.
For the explicit link, you'd probably be able to use `@see {@link QuillOptionsStatic#registry}`, although you'd have to double check me on that.

For the explanation itself, I expected something a little more comprehensive. To the tune of
```
Registry is a data structure provided by the Parchment library.
Registry contains ___. Quill uses a registry to ___.
By default, Quill uses a global registry and you're supposed to configure formats using ___.
This (is broken|doesn't fully prevent Quill from rendering formats) ___ (?).
In the 2.x version of Quill that we are using, there is an undocumented way to provide Quill with a custom registry. Doing so allows ___.

<if there are any limitations or bugs related to custom registries, describe them here as well>

Example usage:
<write a minimal example of creating and configuring a custom registry.>

<finally, here you can link to the commit that introduced custom registries. if you understand how they work under the hood, you can add a brief overview/pointers>
```

If we had control over `Parchment.Registry` type, the initial explanation of what a registry is could've been placed there and we could limit this explanation to just difference between using formats and registries. Since that's not the case, it probably makes sense to fully explain things here.

Reply:
Does `format` option actually not exist at all in quill 2.x?

Reply:
My bad. We do need `format` 🥹